byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import java.util.Collection;
+import java.util.ArrayList;
+
+/**
+ * A class to make a Trip type, containing
+ * a specific id, destinations, interests and routes 
+ * to be used when updating a trip.
+ */
+public final class Trip {
+
+  private final String id;
+  private final String start;
+  private final ArrayList<String> destinations;
+  private final ArrayList<String> interests;
+  private final ArrayList<String> route;
+  public static final String DATASTORE_ENTITY_KIND = "Trip";
+
+  public Trip(String id, String start, Collection<String> destinations,
+              Collection<String> interests, Collection<String> route) {
+    this.id = checkNotNull(id, "id");
+    this.start = checkNotNull(start, "start");
+    Collection<String> validDestinations = checkNotNull(destinations, "destinations");
+    Collection<String> validInterests = checkNotNull(interests, "interests");
+    Collection<String> validRoute = checkNotNull(route, "route");
+    this.destinations = new ArrayList<String>(validDestinations);
+    this.interests = new ArrayList<String>(validInterests);
+    this.route = new ArrayList<String>(validRoute);
+  }
+
+  public ArrayList<String> getInterests() {
+    return new ArrayList<String>(this.interests);
+  }
+
+  public ArrayList<String> getRoute() {
+    return new ArrayList<String>(this.route);
+  }
+
+  public ArrayList<String> getDestinations() {
+    return new ArrayList<String>(this.destinations);
+  }
+
+  public String getTripId() {
+    return this.id;
+  }
+
+  public String getStart() {
+    return this.start;
+  }
+
+  public static Trip createTripFromEntity(Entity tripEntity) {
+    Entity validEntity = checkNotNull(tripEntity, "Trip entity is null");
```

Comment:
I added it here because the entity itself could be null. In the other cases, if it tries to get a property and that property itself is null, it would return null because it's missing the property. But, if the entity itself is null and it tries to get a property, it'd return null for the wrong reason since it'd indicate a property was missing. I removed it just in case I'm overthinking it, but that was my rationale behind that check

Reply:
Yes, having this check is fine, but there is no need to assign the result of `checkNotNull` to a new variable. Here and in `User`.

Reply:
Updated